zokrates_core/src/static_analysis/zir_propagation.rs

```diff
@@ -170,6 +168,10 @@ impl<'ast, T: Field> Folder<'ast, T> for ZirPropagator<'ast, T> {
                 let consequence = self.fold_field_expression(consequence);
                 let alternative = self.fold_field_expression(alternative);
 
+                if consequence.eq(&alternative) {
```

Comment:
I think a more functional style would be better here, using a match statement on consequence and alternative also

```
match (condition, consequence, alternative) {
   (_, conequence, alternative) if consequence == alternative => consequence,
   (true, consequence, _) => consequence,
   (false, _, alternative) => alternative
}
```